tcell_compat.go

```diff
@@ -0,0 +1,524 @@
+func fixColor(c Attribute) tcell.Color {
+	c = c & AttrColorBits
+	// Default color is 0 in tcell/v2 and was 0 in termbox-go, so we are good here
+	if c == ColorDefault {
+		return tcell.ColorDefault
+	}
+
+	tc := tcell.ColorDefault
+	// Check if we have valid color
+	if c&AttrIsValidColor != 0 {
+		tc = tcell.Color(c)
+	} else if c > 0 && c <= 256 {
+		// It's not valid color, but it has value in range 1-256
+		// This is old Attribute style of color from termbox-go (black=1, etc.)
+		// convert to tcell color (black=0|ColorValid)
+		tc = tcell.Color(c-1) | tcell.ColorValid
+	}
+
+	switch outMode {
+	case OutputTrue:
+		return tc
+	case OutputNormal:
+		tc &= tcell.Color(0xf) | tcell.ColorValid
+	case Output256:
+		tc &= tcell.Color(0xff) | tcell.ColorValid
+	case Output216:
+		tc &= tcell.Color(0xff)
+		if tc > 215 {
+			return tcell.ColorDefault
+		}
+		tc += tcell.Color(16) | tcell.ColorValid
+	case OutputGrayscale:
+		tc &= tcell.Color(0x1f)
+		if tc > 26 {
+			return tcell.ColorDefault
+		}
+		tc = grayscale[tc] | tcell.ColorValid
+	default:
+		return tcell.ColorDefault
+	}
+	return tc
+}
+
+func mkStyle(fg, bg Attribute) tcell.Style {
+	st := tcell.StyleDefault
+
+	// extract colors and attributes
+	if fg != ColorDefault {
+		st = st.Foreground(fixColor(fg))
+		st = setAttr(st, fg)
+	}
+	if bg != ColorDefault {
+		st = st.Background(fixColor(bg))
+		st = setAttr(st, bg)
+	}
+
+	return st
+}
+
+func setAttr(st tcell.Style, attr Attribute) tcell.Style {
+	if attr&AttrBold != 0 {
+		st = st.Bold(true)
+	}
+	if attr&AttrUnderline != 0 {
+		st = st.Underline(true)
+	}
+	if attr&AttrReverse != 0 {
+		st = st.Reverse(true)
+	}
+	if attr&AttrBlink != 0 {
+		st = st.Blink(true)
+	}
+	if attr&AttrDim != 0 {
+		st = st.Dim(true)
+	}
+	if attr&AttrItalic != 0 {
+		st = st.Italic(true)
+	}
+	return st
+}
+
+// GetColor creates a Color from a color name (W3C name). A hex value may
+// be supplied as a string in the format "#ffffff".
+func GetColor(color string) Attribute {
+	return Attribute(tcell.GetColor(color))
+}
+
+// Get256Color creates Attribute which stores ANSI color (0-255)
+func Get256Color(color int32) Attribute {
+	return Attribute(color) | AttrIsValidColor
+}
+
+// GetRGBColor creates Attribute which stores RGB color.
+// Color is passed as 24bit RGB value, where R << 16 | G << 8 | B
+func GetRGBColor(color int32) Attribute {
+	return Attribute(color) | AttrIsValidColor | AttrIsRGBColor
+}
+
+// NewRGBColor creates Attribute which stores RGB color.
+func NewRGBColor(r, g, b int32) Attribute {
+	return Attribute(tcell.NewRGBColor(r, g, b))
+}
+
+// tcellClear clears the screen with the given attributes.
+func tcellClear(fg, bg Attribute) {
+	st := mkStyle(fg, bg)
+	w, h := screen.Size()
+	for row := 0; row < h; row++ {
+		for col := 0; col < w; col++ {
+			screen.SetContent(col, row, ' ', nil, st)
+		}
+	}
+}
+
+// InputMode is not used.
+type InputMode int
+
+// Unused input modes; here for compatibility.
```

Comment:
Same for this one

Reply:
Hmm... comment could be updated, but I'm thinking that maybe it should be private too? It's not really exposed as the only input mode change is thru `gocui.Mouse=true`.

Reply:
Hmm oke in that's maybe better

Reply:
I moved the stuff around, this is not used anymore.